### PR TITLE
Bpf filter fix

### PR DESCRIPTION
**Pull request by regit**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3346

Describe changes:
- Fix memory leak
- Fix BPF filter on command line

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output: 
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/507
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/283

### CODE REVIEW

**Comment by victorjulien** [comment]
@dbcfd could you check if this makes sense to you?

File: src/source-pcap-file.c [hunk 1]
```diff
@@ -263,10 +263,9 @@ TmEcode ReceivePcapFileThreadInit(ThreadVars *tv, const void *initdata, void **d
             SCReturnInt(TM_ECODE_OK);
         }
 
+        pv->shared = &ptv->shared;
```
Comment (dbcfd) [inline src/source-pcap-file.c]: I think this causes a double free on shared when status is not ok.

Comment (victorjulien) [reply src/source-pcap-file.c]: @regit can you check this?

Comment (regit) [reply src/source-pcap-file.c]: When build and run with ASAN, it does not show a double free when inverting the clause (and log messages shows we are taking the error path).

When looking at the code, we have 2 cleaning functions call serially before returning:
- CleanupPcapFileFileVars: just set shared to NULL in pcap file strcuture
- CleanupPcapFileThreadVars: do some cleaning inside shared structure

So I think we are good.

Comment (dbcfd) [reply src/source-pcap-file.c]: Works for me. Good to merge.

**Comment by victorjulien** [comment]
Merged as part of #4393, thanks @regit and @dbcfd !

**Pull request closed without merging**